Deploy on paste (for Mac OS only at the moment)

pranaygp opened the pull request:
Closes #482 

This is temporary:
![demo](https://kapture-gvyerjzmxv.now.sh)

I'm still awaiting designs from @evilrabbit and will update it before we ship this.

pranaygp commented:
This now works with png-data (like screenshots taken with <kbd>Cmd + Ctrl + Shift + 4</kbd> on Mac OS)

![](https://kapture-jzwpxbpgoq.now.sh)

pranaygp commented:
Now GIF data in the clipboard (such as the ones exported by `kap` into clipboard) and HTML data (like if you copy code from VSCode) are supported

![](https://file-ywyuoldith.zeit.sh)

evilrabbit commented:
@pranaygp I think the icon is a little bit cut.

pranaygp commented:
@evilrabbit I didn't modify the svg 🤔
Also, it looks exactly like this afaict:
![image](https://user-images.githubusercontent.com/1797812/40952064-b5a27c8c-682e-11e8-9b6d-681842b71036.png)

pranaygp commented:
ok, fixed.
@evilrabbit it looks like this now:
![](https://file-roccowjnby.zeit.sh)

rauchg commented:
How does the tip work? If I install Now Desktop from scratch I'll see the tip right away?

leo commented on renderer/pages/feed.js:
```diff
@@ -407,6 +407,7 @@ class Feed extends Component {
     }
 
     const { getConfig } = this.remote.require('./utils/config')
+    const { deploy } = this.remote.require('./utils/deploy-from-clipboard.js')
```
Let's remove the `.js` here?

pranaygp commented:
@leo I don't think the failing tests are because of this PR (although, in the process of debugging, I ended up including a test for tips to `test-integration`)

When running `yarn test-integration` locally, it goes through all the steps and when the events feed is loaded, there is no "Logged in from Now Desktop" event like this:
![image](https://user-images.githubusercontent.com/1797812/41064627-f7b67c00-6990-11e8-87e2-c461a56c3593.png)

which is exactly what the failing tests are looking for, and so that makes sense

pranaygp commented:
Okay, I think I know what's happening (again, not an issue from this PR but I've fixed it nonetheless @leo). When logging in with `now-desktop-{random garbage}@zeit.pub`, a real account isn't generated and so the event `you logged in fromNow Desktop on macOS` doesn't appear causing the tests to fail. I switched it so every test is run with `now-desktop@zeit.pub` which is a real account and logs those events instead of random email addresses each time.

pranaygp commented:
Related: https://github.com/zeit/now-desktop/pull/462/files#diff-910eb6f57886ca16c136101fb1699231

leo commented:
Merged. We need to wait until the APIs for the event stream are working properly again before releasing this awesome piece!
